pkg/agent/agent.go

```diff
@@ -98,7 +99,14 @@ func (a *Agent) Run(ctx context.Context) error {
 	}
 
 	if a.c.AdminBindAddress != nil {
-		adminEndpoints := a.newAdminEndpoints(manager)
+		// TODO(Mauricio): Avoid creating two different instances of the workload attestor
```

Comment:
As a note, I don't _think_ this will be a problem since workload attestors are designed to handle concurrency.

Reply:
Would be better to create only one `workload_attestor`? ... outside `newEndpoints` and use the same `workload_attestor` for both functions: `newEndpoints()` and `newAdminEndpoints()`. Does it make sense? I think that was the Mauricio's ideia. I did a very small modification locally and worked fine, I don't know if have any impact doing this.

https://github.com/spiffe/spire/blob/830fa6a92d1ba68e449df0e3c51bf2e9b29cff3a/pkg/agent/agent.go#L237-L244

Reply:
The workload attestor component doesn't really carry around state, so it won't make much difference, practically speaking. Conceptually I think it feels cleaner to create it once and give it to both the standard and admin endpoints.

Reply:
Hey @azdagron, sound to me... so I'm going to do that and send it in the next commit, tks :)